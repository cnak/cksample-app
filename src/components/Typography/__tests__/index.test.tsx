import { mount } from 'enzyme';
import React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';

import { BR, BulletPoint, H1, H2, H3, H4, Label, Link, P } from '..';

const onPressSpy = jest.fn();
const mountedLink = mount(<Link onPress={onPressSpy}>Add Something</Link>);

describe('Typography:', () => {
    describe('<H1>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<H1>This is a test</H1>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<H2>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<H2>This is a test</H2>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<H3>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<H3>This is a test</H3>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<H4>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<H4>This is a test</H4>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<P>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<P>This is a test</P>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<BR>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<BR />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<Link>', () => {
        describe('component composition', () => {
            it('includes TouchableOpacity', () => {
                expect(mountedLink.find(TouchableOpacity).length).toEqual(1);
            });
        });

        describe('component methods', () => {
            it('passes onPress function to Touchable', () => {
                expect(onPressSpy).not.toHaveBeenCalled();
                // @ts-ignore
                mountedLink
                    .find(TouchableOpacity)
                    .props()
                    .onPress({} as any);
                expect(onPressSpy).toHaveBeenCalledTimes(1);
            });
        });

        it('matches snapshot', () => {
            const tree = renderer.create(<Link onPress={onPressSpy}>Add Something</Link>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<Label>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<Label>This is a test</Label>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('<BulletPoint>', () => {
        it('matches snapshot', () => {
            const tree = renderer.create(<BulletPoint>This is a test</BulletPoint>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
