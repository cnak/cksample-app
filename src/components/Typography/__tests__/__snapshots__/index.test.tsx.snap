// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography: <BR> matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "height": 16,
        "paddingBottom": 15,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 15,
      },
    ]
  }
/>
`;

exports[`Typography: <BulletPoint> matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#0B0C0C",
          "fontFamily": "Helvetica",
          "fontSize": 16,
          "lineHeight": 20,
          "marginBottom": 15,
        },
      ]
    }
  >
    â€¢
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#0B0C0C",
          "fontFamily": "Helvetica",
          "fontSize": 16,
          "lineHeight": 20,
          "marginBottom": 15,
        },
        Object {
          "flex": 1,
          "paddingLeft": 10,
        },
      ]
    }
  >
    This is a test
  </Text>
</View>
`;

exports[`Typography: <H1> matches snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0B0C0C",
        "fontFamily": "Helvetica",
        "fontSize": 32,
        "fontWeight": "bold",
        "lineHeight": 35,
        "marginBottom": 20,
      },
    ]
  }
>
  This is a test
</Text>
`;

exports[`Typography: <H2> matches snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0B0C0C",
        "fontFamily": "Helvetica",
        "fontSize": 24,
        "fontWeight": "bold",
        "lineHeight": 25.000080000000004,
        "marginBottom": 15,
      },
    ]
  }
>
  This is a test
</Text>
`;

exports[`Typography: <H3> matches snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0B0C0C",
        "fontFamily": "Helvetica",
        "fontSize": 18,
        "fontWeight": "bold",
        "lineHeight": 27,
        "marginBottom": 15,
      },
    ]
  }
>
  This is a test
</Text>
`;

exports[`Typography: <H4> matches snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0B0C0C",
        "fontFamily": "Helvetica",
        "fontSize": 16,
        "lineHeight": 30,
        "marginBottom": 15,
      },
    ]
  }
>
  This is a test
</Text>
`;

exports[`Typography: <Label> matches snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0B0C0C",
        "fontFamily": "Helvetica",
        "fontSize": 16,
        "lineHeight": 20,
        "marginBottom": 10,
      },
    ]
  }
>
  This is a test
</Text>
`;

exports[`Typography: <Link> matches snapshot 1`] = `
<View
  accessibilityHint=""
  accessibilityLabel="Add Something."
  accessibilityRole="link"
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#005EA5",
          "fontFamily": "Helvetica",
          "fontSize": 16,
          "lineHeight": 30,
          "marginBottom": 5,
          "textDecorationColor": "black",
          "textDecorationLine": "underline",
          "textDecorationStyle": "solid",
        },
      ]
    }
  >
    Add Something
  </Text>
</View>
`;

exports[`Typography: <P> matches snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0B0C0C",
        "fontFamily": "Helvetica",
        "fontSize": 16,
        "lineHeight": 20,
        "marginBottom": 15,
      },
    ]
  }
>
  This is a test
</Text>
`;
